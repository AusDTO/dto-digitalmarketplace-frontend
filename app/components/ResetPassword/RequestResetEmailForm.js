import React from 'react'
import PropTypes from 'prop-types'
import { Form } from 'react-redux-form'
import ErrorBox from '../../components/shared/form/ErrorBox'
import Textfield from '../../components/shared/form/Textfield'
import PageAlert from '@gov.au/page-alerts'
import { required, validEmail } from '../validators'

export const RequestResetEmailForm = ({ model, form, user, submitClicked, handleSubmit }) => {
  let { resetPasswordEmailFailure, resetPasswordEmailSuccess } = user
  let { valid } = form

  let hasFocused = false
  const setFocus = e => {
    if (!hasFocused) {
      hasFocused = true
      e.focus()
    }
  }

  return (
    <div className="row">
      <div className="col-sm-push-2 col-sm-8 col-xs-12">
        <article role="main">
          {resetPasswordEmailFailure &&
            valid &&
            <PageAlert as="error">
              <h4>We were unable to send your password reset email</h4>
              <span>
                <p>
                  Please try again later or{' '}
                  <a href="/contact-us" target="_blank" rel="external">
                    {' '}contact us{' '}
                  </a>{' '}
                  for assistance.
                </p>
              </span>
            </PageAlert>}
          {resetPasswordEmailSuccess &&
            <PageAlert as="success">
              <span>
                If the email address you&#39;ve entered belongs to a Marketplace account, we&#39;ll send a link to reset
                the password.
              </span>
            </PageAlert>}
          <ErrorBox model={model} submitClicked={submitClicked} setFocus={setFocus} />
          <header className="page-heading page-heading-without-breadcrumb">
            <h1 className="uikit-display-5">Reset password</h1>
            <span>
              Enter your email address and we&#39;ll send you a link to reset your password. Password reset links are
              valid for 24 hours.
            </span>
          </header>
          <Form model={model} id="sendResetEmail" onSubmit={model => handleSubmit(model)}>
            <Textfield
              model={`${model}.email_address`}
              name="email_address"
              id="email_address"
              htmlFor="email_address"
              description="Enter the email address you used to register with the Marketplace"
              validators={{ required, validEmail }}
              messages={{
                required: 'Your email is required',
                validEmail: 'A valid email is required.'
              }}
            />
            <p>
              <input className="uikit-btn" type="submit" value="Send reset password" onClick={submitClicked} />
            </p>
          </Form>
        </article>
      </div>
    </div>
  )
}
