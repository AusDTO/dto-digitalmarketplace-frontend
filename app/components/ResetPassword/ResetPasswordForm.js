import React from 'react'
import { Form } from 'react-redux-form'
import ErrorBox from '../../components/shared/form/ErrorBox'
import Textfield from '../../components/shared/form/Textfield'
import PageAlert from '@gov.au/page-alerts'
import { passwordLength } from '../validators'
import { rootPath } from '../../routes'

export const ResetPasswordForm = ({ model, user, submitClicked, handleSubmit }) => {
  let { resetPasswordSuccess, getResetDataSuccess, errorMessage } = user

  let hasFocused = false
  const setFocus = e => {
    if (!hasFocused) {
      hasFocused = true
      e.focus()
    }
  }

  return (
    <div className="row">
      <div className="col-sm-push-2 col-sm-8 col-xs-12">
        <article role="main">
          {(resetPasswordSuccess === false || getResetDataSuccess === false) &&
            <PageAlert as="error">
              <h4>We are unable to reset your password</h4>
              <span>
                {errorMessage}
                {getResetDataSuccess === false &&
                  <span>
                    Try <a href={`${rootPath}/reset-password`}> resending </a>your reset password email
                  </span>}
              </span>
            </PageAlert>}
          {resetPasswordSuccess &&
            <PageAlert as="success">
              <h4>You have successfully changed your password</h4>
              <span>
                <p>
                  Please <a href="/login"> login </a> to continue.
                </p>
              </span>
            </PageAlert>}
          <ErrorBox model={model} submitClicked={submitClicked} setFocus={setFocus} />
          <header className="page-heading page-heading-without-breadcrumb">
            <h1 className="uikit-display-5">Reset password</h1>
          </header>
          <Form
            model={model}
            id="resetPassword"
            validators={{
              '': {
                // Form-level validator
                passwordsMatch: val => passwordsMatch(val)
              }
            }}
            onSubmit={model => handleSubmit(model)}
          >
            <Textfield
              model={`${model}.password`}
              name="password"
              id="password"
              htmlFor="password"
              label="Password"
              type="password"
              description="Your password should be at least 10 characters"
              validators={{ passwordLength }}
              messages={{
                passwordLength: 'Your password should be at least 10 characters'
              }}
            />

            <Textfield
              model={`${model}.confirmPassword`}
              name="confirmPassword"
              id="confirmPassword"
              htmlFor="confirmPassword"
              label="Password"
              type="password"
              description="Repeat password used above"
              validators={{
                passwordLength
              }}
              messages={{
                passwordLength: 'Your password confirmation should be at least 10 characters. ',
                passwordsMatch: 'Passwords do not match.'
              }}
            />
            <p>
              <input className="uikit-btn" type="submit" value="Reset password" onClick={submitClicked} />
            </p>
          </Form>
        </article>
      </div>
    </div>
  )
}
