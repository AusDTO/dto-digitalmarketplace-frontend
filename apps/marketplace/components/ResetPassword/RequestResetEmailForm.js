import React from 'react'
import PropTypes from 'prop-types'
import { Form } from 'react-redux-form'
import AUpageAlert from '@gov.au/page-alerts/lib/js/react.js'
import ErrorBox from 'shared/form/ErrorBox'
import Textfield from 'shared/form/Textfield'
import { required, validEmail } from '../validators'

const RequestResetEmailForm = props => {
  const { model, user, submitClicked, handleSubmit } = props
  const { resetPasswordEmailSuccess } = user

  let hasFocused = false
  const setFocus = e => {
    if (!hasFocused) {
      hasFocused = true
      e.focus()
    }
  }

  return (
    <div className="row">
      <div className="col-sm-push-2 col-sm-8 col-xs-12">
        <article role="main">
          {resetPasswordEmailSuccess
            ? <AUpageAlert as="success">
                <span>
                  We have sent a link to reset the password.<br />
                  Check your spam folder if it does not arrive.
                </span>
              </AUpageAlert>
            : <ErrorBox
                title="There was a problem sending your reset email"
                model={model}
                submitClicked={submitClicked}
                setFocus={setFocus}
              />}
          <header className="page-heading page-heading-without-breadcrumb">
            <h1 className="au-display-xl">Reset password</h1>
            <p>
              Enter your email address and we&#39;ll send you a link to reset your password. Password reset links are
              valid for 24 hours.
            </p>
          </header>
          <Form model={model} id="sendResetEmail" onSubmit={data => handleSubmit(data)}>
            <Textfield
              model={`${model}.email_address`}
              name="email_address"
              id="email_address"
              htmlFor="email_address"
              description="Enter the email address you used to register with the Marketplace"
              validators={{ required, validEmail }}
              messages={{
                required: 'Your email is required',
                validEmail: 'A valid email is required.'
              }}
            />
            <p>
              <input className="au-btn" type="submit" value="Send reset password" onClick={submitClicked} />
            </p>
          </Form>
        </article>
      </div>
    </div>
  )
}

RequestResetEmailForm.propTypes = {
  model: PropTypes.string.isRequired,
  user: PropTypes.shape({
    resetPasswordEmailSuccess: PropTypes.bool
  }).isRequired,
  submitClicked: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired
}

export default RequestResetEmailForm
