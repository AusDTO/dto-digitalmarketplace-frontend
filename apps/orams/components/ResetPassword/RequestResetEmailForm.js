import React from 'react'
import PropTypes from 'prop-types'
import { Form } from 'react-redux-form'
import AUpageAlert from '@gov.au/page-alerts/lib/js/react.js'
import ErrorBox from '../../../shared/form/ErrorBox'
import Textfield from '../../../shared/form/Textfield'
import { required, validEmail } from '../../../shared/validators'

import styles from './ResetPassword.scss'

const RequestResetEmailForm = props => {
  const { model, user, submitClicked, handleSubmit } = props
  const { resetPasswordEmailSuccess } = user

  let hasFocused = false
  const setFocus = e => {
    if (!hasFocused) {
      hasFocused = true
      e.focus()
    }
  }

  return (
    <main>
      <div className="row">
        <div className="col-sm-push-2 col-sm-8 col-xs-12">
          <article role="main">
            {resetPasswordEmailSuccess
              ? <AUpageAlert as="success">
                  <span>
                    If the email address you&#39;ve entered belongs to an ORAMS account, we&#39;ll send a link to reset
                    the password. Check your spam folder if it does not arrive.
                  </span>
                </AUpageAlert>
              : <ErrorBox
                  title="There was a problem sending your reset email"
                  model={model}
                  submitClicked={submitClicked}
                  setFocus={setFocus}
                />}
            <header className="page-heading page-heading-without-breadcrumb">
              <h1 className="au-display-xl">Reset password</h1>
              <span>
                Enter your email address and we&#39;ll send you a link to reset your password. Password reset links are
                valid for 24 hours.
              </span>
            </header>
            <Form model={model} id="sendResetEmail" onSubmit={data => handleSubmit(data)}>
              <Textfield
                model={`${model}.email_address`}
                name="email_address"
                id="email_address"
                htmlFor="email_address"
                label="Enter your email address registered to the ORAMS Portal"
                validators={{ required, validEmail }}
                messages={{
                  required: 'Your email is required',
                  validEmail: 'A valid email is required.'
                }}
              />
              <p className={styles.buttonWrapper}>
                <input className="au-btn" type="submit" value="Send reset password" onClick={submitClicked} />
              </p>
            </Form>
          </article>
        </div>
      </div>
    </main>
  )
}

RequestResetEmailForm.propTypes = {
  model: PropTypes.string.isRequired,
  user: PropTypes.shape({
    resetPasswordEmailSuccess: PropTypes.bool
  }).isRequired,
  submitClicked: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired
}

export default RequestResetEmailForm
