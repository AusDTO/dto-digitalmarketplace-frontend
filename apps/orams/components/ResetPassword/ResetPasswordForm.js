import React from 'react'
import PropTypes from 'prop-types'
import { Form } from 'react-redux-form'
import AUpageAlert from '@gov.au/page-alerts/lib/js/react.js'
import ErrorBox from '../../../shared/form/ErrorBox'
import Textfield from '../../../shared/form/Textfield'
import { passwordLength } from '../../../shared/validators'
import { rootPath } from '../../routes'

import styles from './ResetPassword.scss'

const ResetPasswordForm = props => {
  const { model, user, submitClicked, handleSubmit } = props
  const { resetPasswordSuccess } = user

  let hasFocused = false
  const setFocus = e => {
    if (!hasFocused) {
      hasFocused = true
      e.focus()
    }
  }

  return (
    <main>
      <div className="row">
        <div className="col-sm-push-2 col-sm-8 col-xs-12">
          <article role="main">
            {resetPasswordSuccess
              ? <AUpageAlert as="success">
                  <h4>You have successfully changed your password</h4>
                  <span>
                    <p>
                      Please <a href={`${rootPath}/login`}> login </a> to continue.
                    </p>
                  </span>
                </AUpageAlert>
              : <ErrorBox
                  title="There was a problem resetting your password"
                  model={model}
                  submitClicked={submitClicked}
                  setFocus={setFocus}
                />}
            <header className="page-heading page-heading-without-breadcrumb">
              <h1 className="au-display-xl">Reset password</h1>
            </header>
            <Form
              model={model}
              id="resetPassword"
              validators={{
                '': {
                  // Form-level validator
                  passwordsMatch: vals => vals.password === vals.confirmPassword
                }
              }}
              validateOn="submit"
              onSubmit={data => handleSubmit(data)}
            >
              <Textfield
                model={`${model}.password`}
                name="password"
                id="password"
                htmlFor="password"
                label="Password"
                type="password"
                description="Your password should be at least 10 characters"
                validators={{ passwordLength }}
                messages={{
                  passwordLength: 'Your password should be at least 10 characters'
                }}
              />
              <Textfield
                model={`${model}.confirmPassword`}
                name="confirmPassword"
                id="confirmPassword"
                htmlFor="confirmPassword"
                label="Password"
                type="password"
                description="Repeat password used above"
                validators={{
                  passwordLength
                }}
                messages={{
                  passwordLength: 'Your password confirmation should be at least 10 characters. ',
                  passwordsMatch: 'Passwords do not match.'
                }}
              />
              <p className={styles.buttonWrapper}>
                <input className="au-btn" type="submit" value="Reset password" onClick={submitClicked} />
              </p>
            </Form>
          </article>
        </div>
      </div>
    </main>
  )
}

ResetPasswordForm.defaultProps = {
  submitClicked: null,
  handleSubmit: null
}

ResetPasswordForm.propTypes = {
  user: PropTypes.shape({
    resetPasswordSuccess: PropTypes.bool
  }).isRequired,
  model: PropTypes.string.isRequired,
  submitClicked: PropTypes.func,
  handleSubmit: PropTypes.func
}

export default ResetPasswordForm
