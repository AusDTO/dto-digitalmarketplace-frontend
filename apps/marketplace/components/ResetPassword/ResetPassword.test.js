import React from 'react'
import renderer from 'react-test-renderer'
import expect from 'expect'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import configureStore from 'marketplace/store'
import { Provider } from 'react-redux'

import ResetPasswordForm from './ResetPasswordForm'
import RequestResetEmailForm from './RequestResetEmailForm'

Enzyme.configure({ adapter: new Adapter() })

describe('Test suite for ResetPasswordForm component', () => {
  const state = {
    model: 'resetPasswordForm',
    user: {
      resetPasswordEmailFailure: false,
      resetPasswordEmailSuccess: true
    },
    submitClicked: () => true,
    handleSubmit: () => true
  }

  it('CreateUserPage renders without errors', () => {
    const store = configureStore(state)
    const wrapper = shallow(
      <Provider store={store}>
        <ResetPasswordForm {...state} />
      </Provider>
    )
    expect(wrapper).toBeDefined()
  })

  it('Because snapshots', () => {
    const store = configureStore(state)
    const component = renderer.create(
      <Provider store={store}>
        <ResetPasswordForm {...state} />
      </Provider>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})

describe('Test suite for RequestResetEmailForm component', () => {
  const state = {
    model: 'resetPasswordForm',
    user: {
      resetPasswordSuccess: false,
      getResetDataSuccess: true,
      errorMessage: null
    },
    submitClicked: () => true,
    handleSubmit: () => true
  }

  it('RequestResetEmailForm renders without errors', () => {
    const store = configureStore(state)
    const wrapper = shallow(
      <Provider store={store}>
        <RequestResetEmailForm {...state} />
      </Provider>
    )
    expect(wrapper).toBeDefined()
  })

  it('Because snapshots', () => {
    const store = configureStore(state)
    const component = renderer.create(
      <Provider store={store}>
        <RequestResetEmailForm {...state} />
      </Provider>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
