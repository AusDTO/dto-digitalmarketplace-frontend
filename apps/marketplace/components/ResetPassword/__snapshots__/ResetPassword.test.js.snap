// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suite for RequestResetEmailForm component Because snapshots 1`] = `
<div
  className="row"
>
  <div
    className="col-sm-push-2 col-sm-8 col-xs-12"
  >
    <article
      role="main"
    >
      <header
        className="page-heading page-heading-without-breadcrumb"
      >
        <h1
          className="au-display-xl"
        >
          Reset password
        </h1>
        <p>
          Enter your email address and we'll send you a link to reset your password. Password reset links are valid for 24 hours.
        </p>
      </header>
      <form
        id="sendResetEmail"
        noValidate={undefined}
        onReset={[Function]}
        onSubmit={[Function]}
      >
        <div
          className="field "
        >
          <label
            className="question-heading au-text-input__label"
            htmlFor="email_address"
            id="email_address-label"
          >
            
          </label>
          <div
            className="au-text-input__hint"
            id="email_address-hint"
          >
            Enter the email address you used to register with the Marketplace
          </div>
          <input
            aria-describedby="email_address-label"
            className="au-text-input au-text-input--block"
            disabled={false}
            id="email_address"
            max={null}
            maxLength={undefined}
            min={null}
            name="email_address"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyPress={[Function]}
            pattern={null}
            placeholder=""
            readOnly={false}
            step={null}
            type="text"
            value=""
          />
        </div>
        <p>
          <input
            className="au-btn"
            onClick={[Function]}
            type="submit"
            value="Send reset password"
          />
        </p>
      </form>
    </article>
  </div>
</div>
`;

exports[`Test suite for ResetPasswordForm component Because snapshots 1`] = `
<div
  className="row"
>
  <div
    className="col-sm-push-2 col-sm-8 col-xs-12"
  >
    <article
      role="main"
    >
      <header
        className="page-heading page-heading-without-breadcrumb"
      >
        <h1
          className="au-display-xl"
        >
          Reset password
        </h1>
      </header>
      <form
        id="resetPassword"
        noValidate={undefined}
        onReset={[Function]}
        onSubmit={[Function]}
      >
        <div
          className="field "
        >
          <label
            className="question-heading au-text-input__label"
            htmlFor="password"
            id="password-label"
          >
            Password
          </label>
          <div
            className="au-text-input__hint"
            id="password-hint"
          >
            Your password should be at least 10 characters
          </div>
          <input
            aria-describedby="password-label"
            className="au-text-input au-text-input--block"
            disabled={false}
            id="password"
            max={null}
            maxLength={undefined}
            min={null}
            name="password"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyPress={[Function]}
            pattern={null}
            placeholder=""
            readOnly={false}
            step={null}
            type="password"
            value=""
          />
        </div>
        <div
          className="field "
        >
          <label
            className="question-heading au-text-input__label"
            htmlFor="confirmPassword"
            id="confirmPassword-label"
          >
            Password
          </label>
          <div
            className="au-text-input__hint"
            id="confirmPassword-hint"
          >
            Repeat password used above
          </div>
          <input
            aria-describedby="confirmPassword-label"
            className="au-text-input au-text-input--block"
            disabled={false}
            id="confirmPassword"
            max={null}
            maxLength={undefined}
            min={null}
            name="confirmPassword"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyPress={[Function]}
            pattern={null}
            placeholder=""
            readOnly={false}
            step={null}
            type="password"
            value=""
          />
        </div>
        <p>
          <input
            className="au-btn"
            onClick={[Function]}
            type="submit"
            value="Reset password"
          />
        </p>
      </form>
    </article>
  </div>
</div>
`;
